src/coreComponents/mesh/generators/VTKUtilities.cpp

```diff
@@ -808,7 +808,7 @@ ensureNoEmptyRank( vtkSmartPointer< vtkDataSet > mesh,
   integer const myRank = MpiWrapper::commRank();
   auto const myPosition =
     LvArray::sortedArrayManipulation::find( donorRanks.begin(), donorRanks.size(), myRank );
-  bool const isDonor = myPosition != donorRanks.size();
+  bool const isDonor = ( myPosition != donorRanks.size() ) && ( donorRanks[myPosition] == myRank );
```

Comment:
Hmm, I did miss it in the review, but should we even be using `sortedArrayManipulation::find()` which does a binary search on non-sorted data? I think linear search via `std::find` is appropriate here.

Reply:
```suggestion
  auto const pos = std::find( donorRanks.begin(), donorRanks.end(), myRank);
  auto const myPosition = std::distance(donorRanks.begin(), pos);
   bool const isDonor = ( myPosition != donorRanks.size() ) && ( *pos == myRank );
```

then ?

Reply:
No need to compare `*pos == myRank` since `std::find` looks for an exact match, not lower bound, and `pos` is not dereferencable when value is not found. More like
```
   auto const pos = std::find( donorRanks.begin(), donorRanks.end(), myRank);
   bool const isDonor = ( pos != donorRanks.end() );
   ...
   if( isDonor )
   {
     auto const myPosition = std::distance(donorRanks.begin(), pos);
     ...
   }
```

Reply:
Done, thanks for the suggestion!